python/cucim/src/cucim/__init__.py

```diff
@@ -31,4 +31,19 @@
 
 """
 
-from .clara import CuImage, __version__, cli
+# Try to import cupy first.
+# If cucim.clara package is imported first, you may see the following error when running on CUDA 10.x (#44)
+#   python3: Relink `/usr/lib/x86_64-linux-gnu/libnccl.so.2.8.3' with `/lib/x86_64-linux-gnu/librt.so.1' for IFUNC symbol `clock_gettime'
+#   Segmentation fault
+try:
+    import cupy
+except ImportError:
+    pass
```

Comment:
Interesting. Do we know why that is?

Reply:
Don't know the exact root cause, but I guess,

This change is related to https://github.com/rapidsai/cucim/issues/44

If the system has only CUDA 10.x library, since cucim.clara part of the library depends on CUDA 11 and different version of runtime, it looks that it can conflict with symbols CUDA-based library that depends on CUDA 10.X

You can see version-related issues from other Github issue:
https://github.com/tensorflow/tensorflow/issues/19375